Realm/Realm/RealmConfiguration.cs

```diff
@@ -137,6 +159,13 @@ internal override Realm CreateRealm(RealmSchema schema)
                 migration.PopulateConfiguration(ref configuration);
             }
 
+            if (ShouldCompactOnLaunch != null)
+            {
+                var handle = GCHandle.Alloc(this);
+                configuration.should_compact_callback = ShouldCompactOnLaunchCallback;
+                configuration.managed_config_handle = GCHandle.ToIntPtr(handle);
```

Comment:
Consider saving a handle to the delegate instead of the entire configuration object.